src/zeroconf/_core.py

```diff
@@ -170,7 +170,11 @@ def __init__(
         :param apple_p2p: use AWDL interface (only macOS)
         """
         if ip_version is None:
-            ip_version = autodetect_ip_version(interfaces)
+            # Default interface is only possible with IPv4
+            if interfaces is InterfaceChoice.Default:
+                ip_version = IPVersion.V4Only
+            else:
+                ip_version = autodetect_ip_version(interfaces)
```

Comment:
Actually just realized that this is entirely unnecessary as `autodetect_ip_version` defaults to `IPVersion.V4Only` if `interfaces` is not a `list` (so also in the `InterfaceChoice.Default` case). It makes things more explicit, but from a behavior perspective things stay the same.